packages/video_player/video_player_avfoundation/darwin/video_player_avfoundation/Sources/video_player_avfoundation/include/video_player_avfoundation/FVPVideoPlayer.h

```diff
@@ -0,0 +1,109 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#if TARGET_OS_OSX
+#import <FlutterMacOS/FlutterMacOS.h>
+#else
+#import <Flutter/Flutter.h>
+#endif
+
+#import <AVFoundation/AVFoundation.h>
+
+#import "FVPDisplayLink.h"
+#import "FVPFrameUpdater.h"
+#import "FVPVideoPlayerPlugin_Test.h"
+
+/// FVPVideoPlayer is responsible for managing video playback using AVPlayer.
+/// It provides methods to control playback, adjust volume, handle seeking, and
+/// notify the Flutter engine about new video frames.
+@interface FVPVideoPlayer ()
+/// The AVPlayerItemVideoOutput associated with this video player.
+@property(readonly, nonatomic, nonnull) AVPlayerItemVideoOutput *videoOutput;
+/// The plugin registrar, to obtain view information from.
+@property(nonatomic, weak, nullable) NSObject<FlutterPluginRegistrar> *registrar;
+/// The CALayer associated with the Flutter view this plugin is associated with, if any.
+@property(nonatomic, readonly, nullable) CALayer *flutterViewLayer;
```

Comment:
Does this need to be public for some reason?

Looking down the list, it seems like everything that was in the implementation file was moved to this header; only the things that actually need to be public should be. The rest should be declared as private properties in the implementation file.

Reply:
Right, answered [here](https://github.com/flutter/packages/pull/8171#discussion_r1865669097)